src/types/next-auth.d.ts

```diff
@@ -1,9 +1,10 @@
 import { JWT } from "next-auth/jwt";
 import { User } from "./user";
+import { SteamUser } from "./steam-user";
 
 declare module "next-auth" {
   interface Session {
-    user: User;
+    user: User | SteamUser;
```

Comment:
The idea of different login providers is to keep the same user model. Please, refer to my comments on backend project

```suggestion
    user: User;
```